tests/unit/metrics/StatsClient.js

```diff
@@ -27,7 +27,7 @@ describe('StatsClient class', () => {
 
     afterEach(() => redisClient.clear(() => {}));
 
-    it('should correctly record a new request', () => {
+    it('should correctly record a new request by default one increment', () => {
```

Comment:
Here, I suggest you add an actual callback argument and call it at the end of `reportNewRequest`'s callback, to make sure it gets called or tests will timeout.